packages/wrangler/src/pages/functions.tsx

```diff
@@ -0,0 +1,95 @@
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+import { FatalError } from "../errors";
+import { logger } from "../logger";
+import { isInPagesCI, ROUTES_SPEC_VERSION } from "./constants";
+import {
+	isRoutesJSONSpec,
+	optimizeRoutesJSONSpec,
+} from "./functions/routes-transformation";
+import { pagesBetaWarning } from "./utils";
+import type { ArgumentsCamelCase, Argv } from "yargs";
+
+type OptimizeRoutesArgs = {
+	"routes-path": string;
+	"output-routes-path": string;
+};
+
+export function OptimizeRoutesOptions(yargs: Argv): Argv<OptimizeRoutesArgs> {
+	return yargs
+		.options({
+			"routes-path": {
+				type: "string",
+				demandOption: true,
+				description: "The location of the _routes.json file",
+			},
+		})
+		.options({
+			"output-routes-path": {
+				type: "string",
+				demandOption: true,
+				description: "The location of the optimized output routes file",
+			},
+		});
+}
+
+export async function OptimizeRoutesHandler({
+	routesPath,
+	outputRoutesPath,
+}: ArgumentsCamelCase<OptimizeRoutesArgs>) {
+	if (!isInPagesCI) {
+		// Beta message for `wrangler pages <commands>` usage
+		logger.log(pagesBetaWarning);
+	}
+
+	let routesFileContents: string;
+	const routesOutputDirectory = path.dirname(outputRoutesPath);
+
+	if (!existsSync(routesPath)) {
+		throw new FatalError(
+			`Oops! File ${routesPath} does not exist. Please make sure --routes-path is a valid file path (for example "/public/_routes.json").`,
+			1
+		);
+	}
+
+	if (!existsSync(routesOutputDirectory)) {
```

Comment:
Might change this to:
```typescript
!existsSync(routesOutputDirectory) || !lstatSync(routesOutputDirectory).isDirectory()
```
In case a file exists at that path. Not a blocker though